src/frontend/src/features/forms/components/message-composer/index.tsx

```diff
@@ -237,7 +237,52 @@ export const MessageComposer = ({ mailboxId, blockNoteOptions, defaultValue, quo
                         templates={activeSignatures} 
                         isLoading={isLoadingSignatures} 
                         mailboxId={mailboxId}
-                        onSignatureChange={onSaveDraft}
+                        onSignatureChange={(signatureId) => {
+                            if (signatureId === null) {
+                                // Remove signature block
+                                editor.removeBlocks(["signature"]);
+                            } else {
+                                const signatureBlock = editor.getBlock('signature');
+                                const newBlock = { 
+                                    id: "signature", 
+                                    type: "signature" as const, 
+                                    props: { 
+                                        templateId: signatureId, 
+                                        mailboxId: mailboxId 
+                                    } 
+                                };
+
+                                if (signatureBlock) {
+                                    // Replace existing signature
+                                    editor.replaceBlocks(
+                                        ["signature"],
+                                        [newBlock] as unknown as PartialMessageComposerBlockSchema[]
+                                    );
+                                } else {
+                                    // Add new signature at the end
+                                    if (editor.document.length === 0) {
+                                        // If document is empty, first add an empty paragraph
+                                        editor.insertBlocks(
+                                            [{ type: "paragraph", content: [{ type: "text", text: "", styles: {} }] }] as unknown as PartialMessageComposerBlockSchema[],
+                                            "",
+                                            "after"
+                                        );
+                                    }
+                                    // Then insert signature after the last block
+                                    editor.insertBlocks(
+                                        [newBlock] as unknown as PartialMessageComposerBlockSchema[],
+                                        editor.document[editor.document.length - 1].id,
+                                        "after"
+                                    );
+                                }
+                            }
+
+                            // Update form value and trigger draft save
+                            form.setValue('signatureId', signatureId, { shouldDirty: true });
+                            if (onSaveDraft) {
+                                onSaveDraft();
+                            }
+                        }}
```

Comment:
Sorry my previous comment was maybe unclear but IMO all the logic in charge to add/remove signature block must stay into `signature-block` but the logic in charge to update form state and persist draft on signature change should be the responsibility to the `MessageComposer`.